### PR TITLE
<ExchangeAccountKeys />

### LINKED ISSUE #93: Implement the <ExchangeAccountKeys /> component
Image

Relates to:
- [x] #89 
- [x] #88

### PR DESCRIPTION
**Pull request by vitalii-ilchenko-ssa-group**
Closes #93

### CODE REVIEW

File: packages/widgets/src/components/ExchangeAccountKeys/ExchangeAccountKeys.tsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        <div css={S.KeyItem}>
+          <Typography variant="h6" color={theme.colors.greyDropdownFocused}>
+            Secret Key
+          </Typography>
+          <Typography
+            className={isLoading ? 'loading' : ''}
+            variant="h5"
+            weight="bold"
+            css={S.SecretKey}>
+            {secretKey ? secretKey : <span>******</span>}
```
Comment (react-developer-ssa-group) [inline packages/widgets/src/components/ExchangeAccountKeys/ExchangeAccountKeys.tsx:51]: If we have secret key, will we show it in this case? And, if no, we replace it with ******?
If yes, I think it's a mistake and we need to hide this secret key

File: packages/widgets/src/components/ExchangeAccountKeys/ExchangeAccountKeys.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+    const deleteButton = getByRole('button', { name: 'Bin' });
+    await user.click(deleteButton);
+    expect(mockOnClick).toBeCalledTimes(1);
+  });
+
+  it('Renders with API key', () => {
+    const { getByText } = render(<StoryComponent onDelete={jest.fn()} />);
+
+    getByText('123456789012345678901234567890');
```
Comment (arthur-kashaiev-ssa-group) [inline packages/widgets/src/components/ExchangeAccountKeys/ExchangeAccountKeys.spec.tsx:24]: Check there's `***` text present in the component

File: packages/widgets/src/components/ExchangeAccountKeys/ExchangeAccountKeys.tsx [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+} from '@ssa-ui-kit/core';
+import { ExchangeAccountKeysProps } from './types';
+import * as S from './styles';
+
+export const ExchangeAccountKeys = ({
+  title,
+  apiKey,
+  secretKey,
+  onDelete,
+  onVisible,
```
Comment (arthur-kashaiev-ssa-group) [inline packages/widgets/src/components/ExchangeAccountKeys/ExchangeAccountKeys.tsx:18]: I would call it `onVisibilityChange`

File: packages/widgets/src/components/ExchangeAccountKeys/ExchangeAccountKeys.tsx [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+          <Typography variant="h5" weight="bold" css={S.NoWrapText}>
+            {apiKey}
+          </Typography>
+        </div>
+        <div css={S.KeyItem}>
+          <Typography variant="h6" color={theme.colors.greyDropdownFocused}>
+            Secret Key
+          </Typography>
+          <Typography
+            className={isLoading ? 'loading' : ''}
```
Comment (arthur-kashaiev-ssa-group) [inline packages/widgets/src/components/ExchangeAccountKeys/ExchangeAccountKeys.tsx:47]: Do we need to have `isLoading` in the component?

**Review by arthur-kashaiev-ssa-group — CHANGES_REQUESTED** [review]

File: packages/widgets/src/components/ExchangeAccountKeys/ExchangeAccountKeys.spec.tsx (hunk 2 above)
Comment (vitalii-ilchenko-ssa-group) [reply packages/widgets/src/components/ExchangeAccountKeys/ExchangeAccountKeys.spec.tsx:24]: Done

File: packages/widgets/src/components/ExchangeAccountKeys/ExchangeAccountKeys.tsx (hunk 3 above)
Comment (vitalii-ilchenko-ssa-group) [reply packages/widgets/src/components/ExchangeAccountKeys/ExchangeAccountKeys.tsx:18]: Done

File: packages/widgets/src/components/ExchangeAccountKeys/ExchangeAccountKeys.tsx (hunk 4 above)
Comment (vitalii-ilchenko-ssa-group) [reply packages/widgets/src/components/ExchangeAccountKeys/ExchangeAccountKeys.tsx:47]: Done

**Comment by lost-pixel-signals** [comment]
| Shot Type       | Count                | Status   |
| -------------   | -------------------- | -------- |
| Added           | 0       | 🟡       |
| Deleted         | 0     | ❌       |
| Regression      | 0  | ❗       |

**Comment by lost-pixel-signals** [comment]
| Shot Type       | Count                | Status   |
| -------------   | -------------------- | -------- |
| Added           | 4       | 🟡       |
| Deleted         | 0     | ❌       |
| Regression      | 7  | ❗       |
    


*🟡 storybook/widgets-exchange-account-keys--default__[w1440px]*: 


| Image |
| --- |
|  |



*🟡 storybook/widgets-exchange-account-keys--default__[w1920px]*: 


| Image |
| --- |
|  |



*🟡 storybook/widgets-exchange-account-keys--default__[w390px]*: 


| Image |
| --- |
|  |



*🟡 storybook/widgets-exchange-account-keys--default__[w900px]*: 


| Image |
| --- |
|  |



<details>
<summary>❗ custom/widgets-meal-nutrients--hint-opened__[w900px] (click to expand)</summary>

| Before | After |
| --- | --- |
|  |  |
</details>



<details>
<summary>❗ custom/widgets-navbar-opened__[w1440px] (click to expand)</summary>

| Before | After |
| --- | --- |
|  |  |
</details>



<details>
<summary>❗ custom/widgets-navbar-opened__[w1920px] (click to expand)</summary>

| Before | After |
| --- | --- |
|  |  |
</details>



<details>
<summary>❗ custom/widgets-progressInfo__[w1440px] (click to expand)</summary>

| Before | After |
| --- | --- |
|  |  |
</details>



<details>
<summary>❗ custom/widgets-progressInfo__[w390px] (click to expand)</summary>

| Before | After |
| --- | --- |
|  |  |
</details>



<details>
<summary>❗ custom/widgets-progressInfo__[w900px] (click to expand)</summary>

| Before | After |
| --- | --- |
|  |  |
</details>



<details>
<summary>❗ storybook/widgets-heart-rate--default__[w1440px] (click to expand)</summary>

| Before | After |
| --- | --- |
|  |  |
</details>

**Comment by lost-pixel-signals** [comment]
| Shot Type       | Count                | Status   |
| -------------   | -------------------- | -------- |
| Added           | 0       | 🟡       |
| Deleted         | 0     | ❌       |
| Regression      | 1  | ❗       |
    


*❗ storybook/utility-responsive-image--responsive-image__[w1920px]*: 


| Before | After |
| --- | --- |
|  |  |

**Comment by lost-pixel-signals** [comment]
| Shot Type       | Count                | Status   |
| -------------   | -------------------- | -------- |
| Added           | 4       | 🟡       |
| Deleted         | 0     | ❌       |
| Regression      | 5  | ❗       |
    


*🟡 storybook/widgets-exchange-account-keys--default__[w1440px]*: 


| Image |
| --- |
|  |



*🟡 storybook/widgets-exchange-account-keys--default__[w1920px]*: 


| Image |
| --- |
|  |



*🟡 storybook/widgets-exchange-account-keys--default__[w390px]*: 


| Image |
| --- |
|  |



*🟡 storybook/widgets-exchange-account-keys--default__[w900px]*: 


| Image |
| --- |
|  |



*❗ custom/widgets-navbar-opened__[w1440px]*: 


| Before | After |
| --- | --- |
|  |  |



*❗ custom/widgets-navbar-opened__[w1920px]*: 


| Before | After |
| --- | --- |
|  |  |



*❗ custom/widgets-progressInfo__[w1440px]*: 


| Before | After |
| --- | --- |
|  |  |



*❗ storybook/widgets-calories--default__[w1440px]*: 


| Before | After |
| --- | --- |
|  |  |



*❗ storybook/widgets-heart-rate--default__[w1440px]*: 


| Before | After |
| --- | --- |
|  |  |

**Review by arthur-kashaiev-ssa-group — APPROVED** [review]

**Comment by lost-pixel-signals** [comment]
| Shot Type       | Count                | Status   |
| -------------   | -------------------- | -------- |
| Added           | 0       | 🟡       |
| Deleted         | 0     | ❌       |
| Regression      | 1  | ❗       |
    


*❗ storybook/utility-responsive-image--responsive-image__[w1920px]*: 


| Before | After |
| --- | --- |
|  |  |

**Comment by lost-pixel-signals** [comment]
| Shot Type       | Count                | Status   |
| -------------   | -------------------- | -------- |
| Added           | 0       | 🟡       |
| Deleted         | 0     | ❌       |
| Regression      | 4  | ❗       |
    


*❗ custom/widgets-meal-nutrients--hint-opened__[w900px]*: 


| Before | After |
| --- | --- |
|  |  |



*❗ custom/widgets-navbar-opened__[w1920px]*: 


| Before | After |
| --- | --- |
|  |  |



*❗ custom/widgets-user-profile--is-opened__[w1920px]*: 


| Before | After |
| --- | --- |
|  |  |



*❗ storybook/widgets-calories--default__[w1440px]*: 


| Before | After |
| --- | --- |
|  |  |

**Comment by lost-pixel-signals** [comment]
| Shot Type       | Count                | Status   |
| -------------   | -------------------- | -------- |
| Added           | 0       | 🟡       |
| Deleted         | 0     | ❌       |
| Regression      | 1  | ❗       |
    


*❗ storybook/utility-responsive-image--responsive-image__[w1920px]*: 


| Before | After |
| --- | --- |
|  |  |

File: packages/widgets/src/components/ExchangeAccountKeys/ExchangeAccountKeys.tsx [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+  Button,
+  Card,
+  CardContent,
+  CardHeader,
+  Icon,
+  Typography,
+} from '@ssa-ui-kit/core';
+import { ExchangeAccountKeysProps } from './types';
+import * as S from './styles';
+import { useState } from 'react';
```
Comment (arthur-kashaiev-ssa-group) [inline packages/widgets/src/components/ExchangeAccountKeys/ExchangeAccountKeys.tsx:12]: Move this up before ./types and ./styles

**Comment by lost-pixel-signals** [comment]
| Shot Type       | Count                | Status   |
| -------------   | -------------------- | -------- |
| Added           | 0       | 🟡       |
| Deleted         | 0     | ❌       |
| Regression      | 4  | ❗       |
    


*❗ custom/widgets-meal-nutrients--hint-opened__[w1440px]*: 


| Before | After |
| --- | --- |
|  |  |



*❗ custom/widgets-progressInfo__[w1440px]*: 


| Before | After |
| --- | --- |
|  |  |



*❗ storybook/widgets-calories--default__[w1440px]*: 


| Before | After |
| --- | --- |
|  |  |



*❗ storybook/widgets-meal-nutrients--default__[w900px]*: 


| Before | After |
| --- | --- |
|  |  |

**Review by arthur-kashaiev-ssa-group — CHANGES_REQUESTED** [review]

**Review by react-developer-ssa-group — APPROVED** [review]

File: packages/widgets/src/components/ExchangeAccountKeys/ExchangeAccountKeys.tsx (hunk 5 above)
Comment (vitalii-ilchenko-ssa-group) [reply packages/widgets/src/components/ExchangeAccountKeys/ExchangeAccountKeys.tsx]: Done

**Comment by lost-pixel-signals** [comment]
| Shot Type       | Count                | Status   |
| -------------   | -------------------- | -------- |
| Added           | 0       | 🟡       |
| Deleted         | 0     | ❌       |
| Regression      | 1  | ❗       |
    


*❗ storybook/utility-responsive-image--responsive-image__[w1920px]*: 


| Before | After |
| --- | --- |
|  |  |

**Comment by lost-pixel-signals** [comment]
| Shot Type       | Count                | Status   |
| -------------   | -------------------- | -------- |
| Added           | 0       | 🟡       |
| Deleted         | 0     | ❌       |
| Regression      | 1  | ❗       |
    


*❗ storybook/utility-responsive-image--responsive-image__[w1920px]*: 


| Before | After |
| --- | --- |
|  |  |

**Comment by lost-pixel-signals** [comment]
| Shot Type       | Count                | Status   |
| -------------   | -------------------- | -------- |
| Added           | 0       | 🟡       |
| Deleted         | 0     | ❌       |
| Regression      | 2  | ❗       |
    


*❗ custom/widgets-navbar-opened__[w1440px]*: 


| Before | After |
| --- | --- |
|  |  |



*❗ storybook/widgets-calories--default__[w1440px]*: 


| Before | After |
| --- | --- |
|  |  |

**Comment by lost-pixel-signals** [comment]
| Shot Type       | Count                | Status   |
| -------------   | -------------------- | -------- |
| Added           | 0       | 🟡       |
| Deleted         | 0     | ❌       |
| Regression      | 2  | ❗       |
    


*❗ custom/widgets-user-profile--is-opened__[w1920px]*: 


| Before | After |
| --- | --- |
|  |  |



*❗ storybook/widgets-calories--default__[w1440px]*: 


| Before | After |
| --- | --- |
|  |  |

**Pull request merged**